test/e2e/azure_test.go

```diff
@@ -234,4 +235,55 @@ var _ = Describe("Workload cluster creation", func() {
 			})
 		})
 	})
+
+	Context("Creating a GPU-enabled cluster", func() {
+		It("with a single control plane node and 1 node", func() {
+			key := "AZURE_NODE_MACHINE_TYPE"
+			// See https://azure.microsoft.com/en-us/pricing/details/virtual-machines/linux/
+			// TODO: figure out how to set Standard_LRS storage and use Standard_NV6 SKUs.
+			// use env var
+			val := string(compute.VirtualMachineSizeTypesStandardNC6sV3)
```

Comment:
what if I don't have quota for this specific VM size and want to run the test locally. What do you think about using a separate env var for the GPU size in the template? for example `AZURE_GPU_NODE_MACHINE_TYPE`

Reply:
Separate env var is the way to go indeed. I'll make that change.